test/javaserver/Requests/HttpRequestParserTest.java

```diff
@@ -37,6 +50,30 @@ public void testGetsHttpVersion() {
         String version = "HTTP/1.1";
         HttpRequestParser requestParser = new HttpRequestParser(request);
 
-        assertEquals(version, requestParser.getProtocol());
+        assertEquals(version, requestParser.protocol());
+    }
+
+    @Test
+    public void testRequestHeaders() {
+        String host = "localhost:5000";
+        request += "Host: " + host + "\r\n";
+        HttpRequestParser requestParser = new HttpRequestParser(request);
+
+        assertEquals(host, requestParser.getHeader("Host"));
```

Comment:
Some places, you are using `assertEquals` whereas some places, you are using `assertThat(..., is(equalTo(...))`.

I would make sure to be consistent about using `assertThat`.